src/main/java/com/mercateo/common/rest/schemagen/link/LinkCreator.java

```diff
@@ -236,11 +236,13 @@ private void addSchemaIfNeeded(Builder builder, Scope method,
         Optional<String> optionalInputSchema = jsonSchemaGenerator.createInputSchema(method,
                 linkFactoryContext.getFieldCheckerForSchema());
         optionalInputSchema.ifPresent(s -> builder.param(SCHEMA_PARAM_KEY, s));
-        Optional<String> mt = detectMediaType(method.getInvokedMethod());
-        if (mt.isPresent() && MediaType.APPLICATION_JSON.equals(mt.get())) {
-            Optional<String> optionalOutputSchema = jsonSchemaGenerator.createOutputSchema(method,
-                    linkFactoryContext.getFieldCheckerForSchema());
-            optionalOutputSchema.ifPresent(s -> builder.param(TARGET_SCHEMA_PARAM_KEY, s));
+        if (linkFactoryContext.getTargetSchemaEnablerForLink().test(method)) {
```

Comment:
just one question here: why check this in an outer if here, while 

jsonSchemaGenerator.createOutputSchema(method,
                        linkFactoryContext.getFieldCheckerForSchema());

already has an optional return type?

Reply:
you are right. done